src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -300,11 +304,29 @@ class PeerActor(
       def update(ns: Seq[BigInt]) = {
         val maxBlockNumber = ns.fold(0: BigInt) { case (a, b) => if (a > b) a else b }
         if (maxBlockNumber > currentMaxBlockNumber) {
-          context become new HandshakedHandler(rlpxConnection, initialStatus, maxBlockNumber).receive
+          context become new HandshakedHandler(rlpxConnection, initialStatus, maxBlockNumber, blockBroadcastActor).receive
         }
       }
     }
 
+    def handleBlockBroadcastActorTerminated: Receive = {
+      case Terminated(actor) if blockBroadcastActor.contains(actor) =>
+        startNewBlockBroadcastActor()
+    }
+
+    private def startNewBlockBroadcastActor() = {
+      blockBroadcastActor.foreach{ previousBlockBroadcastActor =>
+        context unwatch previousBlockBroadcastActor
+      }
+      val newBlockBroadcastActor = context.actorOf(
+        BlockBroadcastActor.props(nodeStatusHolder, self, context.parent, storage),
+        "blockbroadcast"
+      )
+      context watch newBlockBroadcastActor
+      newBlockBroadcastActor ! BlockBroadcastActor.StartBlockBroadcast
+      context become new HandshakedHandler(rlpxConnection, initialStatus, currentMaxBlockNumber, Some(newBlockBroadcastActor)).receive
```

Comment:
is this correct behavior?
`context become new HandshakedHandler` 
`BlockBroadcastActor` termination does not mean that we are disconnected from the peer, I think changing context is not necessary in this case.
Summoning @LukasGasior1